### PR TITLE
doc: matter: Added page describing Matter bridge architecture

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
The new documentation page describing Matter bridge design and architecture of an implementation in NCS was added.

### CODE REVIEW

File: doc/nrf/protocols/matter/overview/bridge.rst [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+Matter Bridge
+#############
+
+.. contents::
+   :local:
+   :depth: 2
+
+The Matter bridge is a device that allows to use non-Matter IoT devices in a Matter fabric together with other native Matter devices.
+The non-Matter devices in such configuration are called bridged devices.
+It is out of the Matter specification scope to define connectivity technologies used by the bridged devices, so they can use any technology (for example Bluetooth LE or Zigbee), as long as bridge device supports it too.
```
Comment (ArekBalysNordic) [inline doc/nrf/protocols/matter/overview/bridge.rst:12]: Maybe it is worth adding here that we operate within the Application Layer only and translate the Matter data model to other Protocol application layers and vice versa, and do not translate protocols themselves within the lower layers.

File: doc/nrf/protocols/matter/overview/bridge.rst [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+Unlike native Matter nodes, the bridged device has dedicated Bridged Device Basic Information cluster enabled on its endpoint to provide basic information about the device.
+
+Bridged devices can be added to bridge or removed from it at runtime, so the list of endpoints exposed by the bridge may be dynamically updated.
+
+Implementation in |NCS|
+***********************
+
+The |NCS| delivers reference implementation of the Matter bridge core that is located in the :file:`samples/matter/common/bridge` directory.
+The core implementation provides basic bridge functionalities and generic API that is not dependent on any specific connectivity technology.
+Beside of the core part, the |NCS| also delivers example `matter_bridge_app` that can be used as a reference for creating own application.
```
Comment (ArekBalysNordic) [inline doc/nrf/protocols/matter/overview/bridge.rst:41]: Link to the app README page?

Comment (kkasperczyk-no) [reply doc/nrf/protocols/matter/overview/bridge.rst:41]: Oh right, I thought `matter_bridge_app` will work as a link, but it didn't. I will fix it.

File: doc/nrf/protocols/matter/overview/bridge.rst [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+The Matter bridge performs translation between Matter data model and the data representation specific for other protocol.
+It is not possible to provide a generic translation between Matter and all IoT protocols, thus the |NCS| implementation supports only Bluetooth LE protocol.
+Nevertheless you can also disable Bluetooth LE support and use the core implementation part to create an application bridging Matter to any connectivity technology.
```
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/overview/bridge.rst:45]:
```suggestion
It is not possible to provide a generic translation between Matter and all other IoT protocols.
The |NCS| application implements supports for the Bluetooth LE protocol, but you can also disable Bluetooth LE support and use the core implementation part to create an application bridging Matter to any connectivity technology.
```

File: doc/nrf/protocols/matter/overview/bridge.rst [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+The bridge application can be customized on various fields depending on the supported connectivity protocol, supported Matter device types for bridging or used UI to control the device.
```
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/overview/bridge.rst:47]:
```suggestion
The bridge application can be customized in various ways depending on the supported connectivity protocol and Matter device types for bridging, or the UI used to control the device.
```

File: doc/nrf/protocols/matter/overview/bridge.rst [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+The following figure illustrates the software components of the bridge core and their relation with the User's application:
```
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/overview/bridge.rst:48]:
```suggestion
The following figure shows the software components of the bridge core and their relation to the user's application:
```

File: doc/nrf/protocols/matter/overview/bridge.rst [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+.. figure:: images/matter_bridge_architecture.svg
+   :alt: Matter bridge software architecture
+
+   Matter bridge software architecture
+
+On the figure above, there are four major software modules included in the Bridge core:
```
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/overview/bridge.rst:55]:
```suggestion
In the figure, there are four major software modules included in the Bridge core:
```

File: doc/nrf/protocols/matter/overview/bridge.rst [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+* ``Matter Bridged Device`` - the abstract class that defines common functionalities for all Matter bridged devices.
+                              The User's application can create specific Matter bridged device objects, for example OnOff Light Device, that inherit the base class functionality.
```
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/overview/bridge.rst:58]:
```suggestion
* ``Matter Bridged Device`` - The abstract class that defines common functionalities for all Matter bridged devices.
  The user's application can create specific Matter bridged device objects, such as OnOff Light Device, that inherit the base class functionality.
```

File: doc/nrf/protocols/matter/overview/bridge.rst [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+* ``Bridged Device Data Provider`` - the abstract class that defines common functionalities for all Bridged Device Data Providers.
+                                     The providers are objects that provide data for Matter bridged devices and represent interface to the bridged device on the non-Matter side.
+                                     The User's application can create specific implementation of a data provider that can, for example obtain data from physical devices using Bluetooth LE protocol.
```
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/overview/bridge.rst:61]:
```suggestion
* ``Bridged Device Data Provider`` - The abstract class that defines common functionalities for all Bridged Device Data Providers.
  The providers are objects that provide data for Matter bridged devices and represents the interface to the bridged device on the non-Matter side.
  The user's application can create a specific implementation of a data provider that can, for example, obtain data from physical devices using the Bluetooth LE protocol.
```

File: doc/nrf/protocols/matter/overview/bridge.rst [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+* ``Bridge Manager`` - the main object of a bridge that forwards information between bridged devices, their data providers, User's application and Matter stack.
```
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/overview/bridge.rst:62]:
```suggestion
* ``Bridge Manager`` - The main object of a bridge.
  It forwards information between bridged devices, their data providers, the user's application, and the Matter stack.
```

File: doc/nrf/protocols/matter/overview/bridge.rst [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+* ``Bridge Storage Manager`` - the software module that is used to persist information about bridged devices connected to the bridge that allows to restore the application state after bridge device reboot.
```
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/overview/bridge.rst:63]:
```suggestion
* ``Bridge Storage Manager`` - The software module that is used to make information about bridged devices connected to the bridge persistent.
  It makes it possible to restore the application state after rebooting the bridge device.
```

File: doc/nrf/protocols/matter/overview/bridge.rst [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+.. contents::
+   :local:
+   :depth: 2
+
+A Matter bridge is a device that makes it possible to use non-Matter IoT devices in a Matter fabric together with native Matter devices.
+A non-Matter devices used this way is called a *bridged device*.
+A Matter bridge exposes the bridged devices to Matter nodes in a fabric as Matter endpoints.
+This exposure is transparent, meaning that the Matter nodes are able to communicate with non-Matter bridged devices as if they were native Matter devices.
+
+It is out of scope for the Matter specification to define connectivity technologies used by the bridged devices, so they can use any technology (for example, Bluetooth® LE or Zigbee) as long as Matter bridge supports it too.
```
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/overview/bridge.rst:15]:
```suggestion
It is out of scope for the Matter specification to define connectivity technologies used by the bridged devices, so they can use any technology (for example, Bluetooth® LE or Zigbee) as long as the Matter bridge supports it too.
```

File: doc/nrf/protocols/matter/overview/bridge.rst [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+The Matter bridge translates between the specific subset of Matter data model and the data representation specific for other protocol.
+It is not the role of a bridge to translate whole traffic between the protocols used on the networking layer.
```
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/overview/bridge.rst:17]:
```suggestion
It is not the role of a bridge to translate all traffic between the protocols used on the networking layer.
```

**Review by wiba-nordic — APPROVED** [review]

**Review by ArekBalysNordic — APPROVED** [review]

File: doc/nrf/protocols/matter/overview/bridge.rst [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+Implementation in the |NCS|
+***************************
+
+The |NCS| includes a reference implementation of the Matter bridge core, located in the :file:`samples/matter/common/bridge` directory.
+The core implementation is not dependent on any specific connectivity technology, and provides basic bridge functionalities and a generic API.
+In addition to the core part, the |NCS| also includes an example :ref:`matter_bridge_app` application that you can use as a reference for creating your own application.
+
+The Matter bridge translates between the Matter data model and the data representation specific for other protocols.
+It is not possible to provide a generic translation between Matter and all other IoT protocols.
+The |NCS| application implements supports for the Bluetooth LE protocol, but you can also disable Bluetooth LE support and use the core implementation part to create an application bridging Matter to any connectivity technology.
```
Comment (markaj-nordic) [inline doc/nrf/protocols/matter/overview/bridge.rst:47]:
```suggestion
The |NCS| application implements support for the Bluetooth LE protocol, but you can also disable Bluetooth LE support and use the core implementation part to create an application bridging Matter to any connectivity technology.
```

File: doc/nrf/protocols/matter/overview/bridge.rst [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+
+The bridge application can be customized in various ways depending on the supported connectivity protocol and Matter device types for bridging, or the UI used to control the device.
```
Comment (markaj-nordic) [inline doc/nrf/protocols/matter/overview/bridge.rst:49]:
```suggestion
The bridge application can be customized in various ways depending on the supported connectivity protocol and Matter device types used for bridging, or the UI used to control the device.
```
?

File: doc/nrf/protocols/matter/overview/bridge.rst [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+   :alt: Matter bridge software architecture
+
+   Matter bridge software architecture
+
+In the figure, there are four major software modules included in the Bridge core:
+
+* ``Matter Bridged Device`` - The abstract class that defines common functionalities for all Matter bridged devices.
+  The user's application can create specific Matter bridged device objects, such as OnOff Light Device, that inherit the base class functionality.
+* ``Bridged Device Data Provider`` - The abstract class that defines common functionalities for all Bridged Device Data Providers.
+  The providers are objects that provide data for Matter bridged devices and represents the interface to the bridged device on the non-Matter side.
```
Comment (markaj-nordic) [inline doc/nrf/protocols/matter/overview/bridge.rst:62]:
```suggestion
  The providers are objects that provide data for Matter bridged devices and represent the interface to the bridged device on the non-Matter side.
```

File: doc/nrf/protocols/matter/overview/bridge.rst [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+  The user's application can create a specific implementation of a data provider that can, for example, obtain data from physical devices using the Bluetooth LE protocol.
+* ``Bridge Manager`` - The main object of a bridge.
```
Comment (markaj-nordic) [inline doc/nrf/protocols/matter/overview/bridge.rst:64]: `Main object` sounds a bit odd to me, maybe it's better to refer to it as a `core component/module`

File: doc/nrf/protocols/matter/overview/images/matter_bridge_architecture.svg
Comment (markaj-nordic) [inline doc/nrf/protocols/matter/overview/images/matter_bridge_architecture.svg:1]: I's aware that this is not really an UML diagram, but are we sure that the relation between user's application and Bridge Storage Manager can be classified as a association? I seems that it's only an application to call BSM API (and not vice versa). So maybe, it's more proper to use one way association (an arrow with thin arrowhead). The same concerns application-Bridge Manager relation.

Also, it seems to me that when it comes to the concrete devices we show only the Matter part, i.e. we might think about depicting the concrete BLE provider which inherits from BLE Bridged Device Provider.

**Review by markaj-nordic — APPROVED** [review]
Overall LGTM, I found some typos however. The comments to the diagram might be treated as a suggestion or a subject of iterative improvement.

**Comment by wiba-nordic** [comment]
Oh, nice catches on the typos!

File: doc/nrf/protocols/matter/overview/images/matter_bridge_architecture.svg
Comment (kkasperczyk-no) [reply doc/nrf/protocols/matter/overview/images/matter_bridge_architecture.svg:1]: To be honest I didn't plan to use UML-related relations, however I used it to present the inheritance, so you are right regarding one way association. I will update it to be consistent.

Regarding BLE devices, as we discussed F2F I will update the image.

File: doc/nrf/protocols/matter/overview/bridge.rst (hunk 16 above)
Comment (kkasperczyk-no) [reply doc/nrf/protocols/matter/overview/bridge.rst:64]: But we described bridge "core", as a group of 4 modules common for all bridge applications, no matter what technologies they use. Would replacing it with principal/chief word work for you?

Comment (wiba-nordic) [reply doc/nrf/protocols/matter/overview/bridge.rst:64]: I agree using "core" for two different meanings on the same page isn't great. Would "main component" work?

Comment (kkasperczyk-no) [reply doc/nrf/protocols/matter/overview/bridge.rst]: I updated it with the "main component".

File: doc/nrf/protocols/matter/overview/bridge.rst [hunk 17]
```diff
@@ -0,0 +1,67 @@
...
+   :depth: 2
+
+A Matter bridge is a device that makes it possible to use non-Matter IoT devices in a Matter fabric together with native Matter devices.
+A non-Matter devices used this way is called a *bridged device*.
+A Matter bridge exposes the bridged devices to Matter nodes in a fabric as Matter endpoints.
+This exposure is transparent, meaning that the Matter nodes are able to communicate with non-Matter bridged devices as if they were native Matter devices.
+
+It is out of scope for the Matter specification to define connectivity technologies used by the bridged devices, so they can use any technology (for example, Bluetooth® LE or Zigbee) as long as the Matter bridge supports it too.
+The Matter bridge translates between the specific subset of the Matter data model and the data representation specific for the other protocol.
+It is not the role of a bridge to translate all traffic between the protocols used on the networking layer.
```
Comment (LuDuda) [inline doc/nrf/protocols/matter/overview/bridge.rst:17]: Maybe, we can state that it is up to product manufacturer (bridge manufacturer) to define which device types on Matter and which bridged devices it supports.

**Review by LuDuda — APPROVED** [review]
Looks fantastic!

**Pull request merged**